### PR TITLE
Feature/aab bundle

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -53,43 +53,45 @@ inputs:
...
+        "*.aab"
         ```
 
         Copy every APK file with a filename that contains `release`, like (`./app/build/outputs/apk/app-release-unsigned.apk`):
 
         ```
         *release*.apk
         ```
-  - apk_file_exclude_filter: |
+  - app_file_exclude_filter: |
```
Comment (BirmacherAkos) [inline step.yml]: Do not remove the `apk_file_exclude_filter`.
The input is deprecated, so put it under the `Deprecated` category, but the step should be backward compatible.

Comment (BirmacherAkos) [reply step.yml]: If you remove the input from the step.yml, the CLI will not parse the input, even if it's provided to the step.

File: main.go [hunk 2]
```diff
@@ -286,6 +212,37 @@ func findDeployPth(deployDir, baseName, ext string) (string, error) {
...
+	if config.GradleFile != "" {
+		if exist, err := pathutil.IsPathExists(config.GradleFile); err != nil {
+			return fmt.Errorf("failed to check if GradleFile exists at: %s, error: %s", config.GradleFile, err)
+		} else if !exist {
+			return fmt.Errorf("GradleFile does not exist at: %s", config.GradleFile)
+		}
+	}
+
+	if config.ApkFileIncludeFilter != "" {
+		log.Warnf("Please use *APK and AAB file include filter* instead of *APK file include filter*.")
```
Comment (godrei) [inline main.go]: i would put more pressure on the user, i used something like this: `step input 'APK file path' (apk_path) is deprecated and will be removed soon, use 'APK or App Bundle file path' (app_path) instead!`

Comment (godrei) [reply main.go]: Other than that, if the user already has a custom filter and the new filter is on default we should use the old filter, to do not break existing configs

File: main.go [hunk 3]
```diff
@@ -428,67 +377,94 @@ func main() {
...
-		if err := command.CopyFile(apkFile, deployPth); err != nil {
+		log.Printf("copy %s to %s", appFile, deployPth)
+		if err := command.CopyFile(appFile, deployPth); err != nil {
 			failf("Failed to copy apk, error: %s", err)
 		}
 
-		lastCopiedApkFile = deployPth
-		copiedApkFiles = append(copiedApkFiles, deployPth)
+		switch strings.ToLower(ext) {
+		case ".apk":
```
Comment (godrei) [inline main.go]: do not define this case if this is the default as well

File: main.go [hunk 4]
```diff
@@ -428,67 +377,94 @@ func main() {
...
+			}
+			log.Donef("The app path is now available in the Environment Variable: $%s (value: %s)", appEnv, lastCopiedFile)
 		}
-		log.Donef("The apk path is now available in the Environment Variable: $BITRISE_APK_PATH (value: %s)", lastCopiedApkFile)
 	}
-	if len(copiedApkFiles) > 0 {
-		apkList := strings.Join(copiedApkFiles, "|")
-		if err := exportEnvironmentWithEnvman("BITRISE_APK_PATH_LIST", apkList); err != nil {
-			failf("Failed to export enviroment (BITRISE_APK_PATH_LIST), error: %s", err)
+	for appListEnv, appFiles := range map[string][]string{
```
Comment (godrei) [inline main.go]: could not it be merged with the above for loop?

File: main.go (hunk 2 above)
Comment (lpusok) [reply main.go]: It should behave like that.

File: main.go (hunk 4 above)
Comment (lpusok) [reply main.go]: No, appending _LIST to the env vars manuallt is hard to read.

File: main.go (hunk 2 above)
Comment (godrei) [reply main.go]: the default include filter is `*.apk`, if the user defined something else, like `*debug*.apk` and the new filter is on default value `*.app\n*.aab`, then use the old filter's value.

File: bitrise.yml [hunk 5]
```diff
@@ -156,72 +187,37 @@ workflows:
...
-            #!/bin/bash
-            set -ex
-
-            # Check for unhandled errors
-            go get -u -v github.com/kisielk/errcheck
-
-            # Go lint
-            go get -u -v github.com/golang/lint/golint
+            if [ ! -f $BITRISE_APK_PATH ]; then
+              echo "AAB not found at ${BITRISE_APK_PATH}"
```
Comment (godrei) [inline bitrise.yml]: "APK not found at ${BITRISE_APK_PATH}"

File: step.yml [hunk 6]
```diff
@@ -53,43 +53,45 @@ inputs:
...
     opts:
       category: Export Config
-      title: "APK file include filter"
+      title: "APK and AAB file include filter"
       description: |
-        The step will copy the generated APK files that match this filter into the Bitrise Deploy Directory.
+        The step will copy the generated APK and AAB files that match this filter into the Bitrise Deploy Directory.
 
-        This filter is a standard find commands -path pattern, see: http://linux.die.net/man/1/find
+        If you specify an empty filter, the step will set this filter to: `*.apk \n *.aab`.
```
Comment (godrei) [inline step.yml]: pls remove this line

**Pull request merged**